src/database/src/js/postGresOperations.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020-2021 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/fluid-project/preferencesServer/blob/main/LICENSE
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    pg    = require("pg");
+
+fluid.registerNamespace("fluid.postgresdb");
+
+fluid.defaults("fluid.postgresdb.request", {
+    gradeNames: ["fluid.component"],
+
+    // The following are set by integrators that use this component
+    database:       null,   // e.g., "fluid_prefsdb"
+    host:           null,   // e.g., "localhost"
+    port:           null,   // e.g., 5432
+    user:           null,   // e.g., "admin"
+    password:       null,
+    members: {
+        // Reference to a node-postgres Pool instance that will handle the
+        // Postgres queries, initialized onCreate.
+        pool:  null
+    },
+    listeners: {
+        "onCreate": {
+            funcName: "fluid.postgresdb.initConnection",
+            args: ["{that}"]
+        }
+    },
+    invokers: {
+        "query": {
+            funcName: "fluid.postgresdb.query",
+            args: ["{that}.pool", "{arguments}.0"]
+                                  // SQL query string
+        },
+        "bulkQuery": {
+            funcName: "fluid.postgresdb.bulkQuery",
+            args: ["{that}", "{arguments}.0"]
+                              // array of SQL query strings
+        },
+        "loadFromJSON" : {
+            funcName: "fluid.postgresdb.loadFromJSON",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+                             // table name    // array of JSON
+        }
+    }
+});
+
+/**
+ * Initialize connection to the postgres database, using node-postgres.
+ *
+ * @param {Object} that - Contains the options for the connection and a member
+ *                        to hold a reference to the connection.
+ * @param {String} that.options.database - Name of the database.
+ * @param {String} that.options.user - Name of the user with admin access.
+ * @param {String} that.options.password - User's password.
+ * @param {Number} that.options.port - The port number for the connection.
+ * @param {Object} that.pool - Allocated and used to connect to the database.
+ *
+ */
+fluid.postgresdb.initConnection = function (that) {
+    that.pool = new pg.Pool({
+        user:       that.options.user,
+        password:   that.options.password,
+        host:       that.options.host,
+        database:   that.options.database,
+        port:       that.options.port
+    });
+};
+
+/**
+ * Convenience function to surface node-postgres `query()` function and log any
+ * errors.
+ *
+ * @param {Object} pool - node-postgres Pool instance to use for querying.
+ * @param {String} queryString - SQL query.
```

Comment:
Be careful with using "SQL query" when what you mean is "SQL statement". As mentioned earlier, the word "query" gives an impression of SELECT statement.

This description probably can be changed to: A string that contains one or more SQL statements.

This comment applies to other places where "SQL query" is used. Thanks.

Reply:
Right, I've adjusted the comments as appropriate.